tempesta_fw/cache.c

```diff
@@ -996,7 +996,7 @@ __tfw_cache_strcpy(char **p, TdbVRec **trec, TfwStr *src, size_t tot_len,
 {
 	long copied = 0;
 
-	while (copied < src->len) {
+	while (copied < src->len && tot_len - copied) {
```

Comment:
The condition is strange and masks issue root. Why a new string to be copied to cache could be bigger than response length in total? This is the question that needs to be fixed.

This workaround messes up the cache entry and the response is _never cached and always forwarded from origin_.